pandas/core/dtypes/cast.py

```diff
@@ -154,6 +155,30 @@ def maybe_box_datetimelike(value: Scalar, dtype: Optional[Dtype] = None) -> Scal
     return value
 
 
+def maybe_box_native(value: Scalar) -> Scalar:
+    """
+    If passed a scalar cast the scalar to a python native type.
+
+    Parameters
+    ----------
+    value : scalar or Series
+
+    Returns
+    -------
+    scalar or Series
+    """
+    if is_datetime_or_timedelta_dtype(value):
```

Comment:
is it just np.datetime64/np.timedelta64 objects being caught here?

Reply:
and also `datetime.datetime` (`datetime.timedelta`) get boxed to `pd.Timestamp` (`pd.Timedelta`)